### PR TITLE
Add more two-step libyuv RGB-to-YUV conversions

### PR DESCRIPTION
**Pull request by y-guyon**
Emulate the following:
RAWToI444
RAWToI422
RAWToJ422
RGB24ToI444
RGB24ToI422
RGB24ToJ422

Use the following:
RAWToJ400
RGB24ToJ400
RGBAToJ400

Change lutRgbToYuv indexing order for clearer code.

### CODE REVIEW

**Review by maryla-uc — APPROVED** [review]

File: src/reformat_libyuv.c [hunk 1]
```diff
@@ -63,6 +63,7 @@ unsigned int avifLibYUVVersion(void)
 
 // These defines are used to create a NULL reference to libyuv functions that
 // did not exist prior to a particular version of libyuv.
+// Versions prior to 1750 are considered too old and not checked.
```
Comment (wantehchang) [inline src/reformat_libyuv.c:66]: This version should match the minimum version of libyuv in CMakeLists.txt, currently 1755:

```
find_package(libyuv QUIET) # not required
if(libyuv_FOUND)
    ...
    if(NOT LIBYUV_VERSION)
        message(STATUS "libavif: libyuv found, but version unknown; libyuv-based fast paths disabled.")
    elseif(LIBYUV_VERSION LESS 1755)
        message(STATUS "libavif: libyuv (${LIBYUV_VERSION}) found, but is too old; libyuv-based fast paths disabled.")
    else()
```

File: src/reformat_libyuv.c [hunk 2]
```diff
@@ -220,12 +235,12 @@ avifResult avifImageRGBToYUVLibYUV8bpc(avifImage * image, const avifRGBImage * r
             typedef int (*RGBtoY)(const uint8_t *, int, uint8_t *, int, int, int);
             // Second dimension is for avifRange.
             RGBtoY lutRgbToY[AVIF_RGB_FORMAT_COUNT][2] = {
-                { NULL, NULL },             // RGB
+                { NULL, RAWToJ400 },        // RGB
```
Comment (wantehchang) [inline src/reformat_libyuv.c:238]: Nit: I don't have a preference, but if would be good to make `avifRange` either the first dimension or the last dimension in both lookup tables.

File: src/reformat_libyuv.c (hunk 1 above)
Comment (y-guyon) [reply src/reformat_libyuv.c:66]: Done at https://github.com/AOMediaCodec/libavif/pull/1081

File: src/reformat_libyuv.c (hunk 2 above)
Comment (y-guyon) [reply src/reformat_libyuv.c:238]: Done at https://github.com/AOMediaCodec/libavif/pull/1082

**Pull request merged**